src/iddata/loader.py

```diff
@@ -303,21 +304,33 @@ def load_ilinet(self,
 
 
   def load_nhsn(self, rates=True, drop_pandemic_seasons=True, as_of=None):
-    if drop_pandemic_seasons:
-      if as_of is None:
-        file_path = "influenza-hhs/hhs.csv"
-      else:
-        # find the largest stored file dated on or before the as_of date
-        as_of_file_path = f"influenza-hhs/hhs-{str(as_of)}.csv"
-        glob_results = s3fs.S3FileSystem(anon=True) \
-            .glob("infectious-disease-data/data-raw/influenza-hhs/hhs-????-??-??.csv")
-        all_file_paths = sorted([f[len("infectious-disease-data/data-raw/"):] for f in glob_results])
-        all_file_paths = [f for f in all_file_paths if f <= as_of_file_path]
-        file_path = all_file_paths[-1]
+    if not drop_pandemic_seasons:
+      raise NotImplementedError("Functionality for loading all seasons of NHSN data with specified as_of date is not implemented.")
+    
+    if as_of is None:
+      as_of = datetime.date.today()
+    
+    if isinstance(as_of, str):
+      as_of = datetime.date.fromisoformat(as_of)
```

Comment:
It wouldn't hurt to try/catch fromisoformat() in case the format is invalid

Reply:
I think it would be better to throw an error if this date is malformatted than to set it to a value the caller may not be expecting.